src/LBPairBeaconProxy.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-3
+pragma solidity 0.8.20;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+
+
+interface IOwnable {
+    function owner() external view returns (address);
+}
+
+/**
+ * @dev this is the Proxy contract of the LBPair.
+ * @dev this contract instantiates LBPair contract via OZ Beacon Proxy (LBPairUpgradeableBeacon in this case).
+ */
+contract LBPairBeaconProxy is BeaconProxy {
+    receive() external payable {}
+
+    constructor(address _beaconAddress, address _tokenX, address _tokenY, uint16 _binStep, bytes memory _data) payable BeaconProxy(_beaconAddress, "") {
```

Comment:
anything speaking against passing the `data` to the BeaconProxy and remove the last line `ERC1967Utils.upgradeBeaconToAndCall(_beaconAddress, _data);` from this function because it will be called from the BeaconProxy constructor anyway?
because otherwise
```suggestion
    constructor(address _beaconAddress, address _tokenX, address _tokenY, uint16 _binStep, bytes memory _data) payable BeaconProxy(_beaconAddress, data) {
```